// chatgpt/src/msft-get-prompt-response-from-models.js
import dotenv from 'dotenv';
import fs from 'fs';
import { url } from 'inspector';
import fetch from "node-fetch";
import getDepolymentName from 'node-fetch/src/get-deployment-name.js';

dotenv.config();

const baseUrl = "https://dalle-exploration.openai.azure.com";
const headers = {
  'api-key': process.env.OPENAI_MSFT_KEY,
  'Content-Type': 'application/json'
};
function getModelUrl(modelName, endPoint='completions'){
  console.log(modelName);
  var deploymentName = getDepolymentName(modelName);
  return `${baseUrl}/openai/deployments/${deploymentName}/${endPoint}?api-version=2022-12-01`  
}

export default async  function msftGetPromptResponseFromModels(prompt, 
                                    {endPoint='completions',
                                    model='text-davinci-002', 
                                    temperature=1, 
                                    max_tokens=400,
                                    top_p=0.6,
                                    frequency_penalty=0,
                                    presence_penalty=0,
                                    best_of=1}){
    const urlEndPoint = getModelUrl(model,  endPoint);
    //console.log(urlEndPoint);
    const bodyObject = {
      model: model,
      prompt: prompt,
      max_tokens: max_tokens,
      temperature: temperature,
      frequency_penalty: frequency_penalty,
      presence_penalty: presence_penalty,
      top_p: top_p,
      best_of: best_of,
    };
    //console.log(bodyObject)
  const response = await fetch(urlEndPoint, {
    method: "POST",
    headers: headers,
    body: JSON.stringify(bodyObject)
  });
  return await response.json();
}
