// chatgpt/src/public-get-prompt-response-from-models.js
import dotenv from 'dotenv';
import fs from 'fs';
import fetch from "node-fetch";

dotenv.config();

const modelsEndpoint = 'https://api.openai.com/v1/models';
const headers = {
  authorization: `Bearer ${process.env.OPENAI_API_KEY}`,
  'Content-Type': 'application/json'
};

export default async  function publicGetPromptResponseFromModels(prompt, 
                                    {endpoint='https://api.openai.com/v1/completions',
                                    model='text-davinci-003', 
                                    temperature=1, 
                                    max_tokens=400,
                                    top_p=0.6,
                                    frequency_penalty=0,
                                    presence_penalty=0,
                                    best_of=1}){
    const bodyObject = {
      model: model,
      prompt: prompt,
      max_tokens: max_tokens,
      temperature: temperature,
      frequency_penalty: frequency_penalty,
      presence_penalty: presence_penalty,
      top_p: top_p,
      best_of: best_of,
    };
    //console.log(bodyObject)
  const response = await fetch(endpoint, {
    method: "POST",
    headers: headers,
    body: JSON.stringify(bodyObject)
  });
  return await response.json();
}
